ql/experimental/math/zigguratrng.cpp

```diff
@@ -137,7 +137,7 @@ namespace QuantLib {
         };
 
         // values of 2^{-24}*x[i]. w_[0] is special.
-        const Real w_[128] = {
+        const double w_[128] = {
```

Comment:
Could these arrays be `constexpr` instead? Any drawbacks?

Reply:
There are no drawbacks making them `constexpr` if they are `double`. In fact, to modernise the code, they could be defined as `constexpr std::array<double, 128>` instead of C arrays. 

There are a lot more similar occurrences in the codebase though - we think that modernising them could be a separate PR. This PR is only fixing the largest constant arrays that cause long compile times if `Real` is not `double`.

Reply:
ok, thanks.

Reply:
I also think it's a good idea to modernize the C arrays, and when we do so then we should remove the following two clang-tidy suppressions as well:

https://github.com/lballabio/QuantLib/blob/master/.clang-tidy#L18
https://github.com/lballabio/QuantLib/blob/master/.clang-tidy#L40